router/pkg/connectrpc/operation_registry.go

```diff
@@ -0,0 +1,289 @@
+package connectrpc
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"go.uber.org/zap"
+
+	"github.com/wundergraph/cosmo/router/pkg/schemaloader"
+
+	"github.com/wundergraph/graphql-go-tools/v2/pkg/ast"
+	"github.com/wundergraph/graphql-go-tools/v2/pkg/astparser"
+)
+
+// OperationRegistry manages pre-defined GraphQL operations for ConnectRPC.
+// Operations are scoped to their service (package.service) and cached in memory
+// for fast access during request handling.
+//
+// Thread-safety: This registry is designed to be safe for concurrent use with
+// multiple readers and occasional writers. Read operations (Get*, Has*, Count*)
+// can be called concurrently, and write operations (Load*, Clear*) use proper
+// locking to ensure consistency.
+type OperationRegistry struct {
+	// Service-scoped operations: serviceName (package.service) -> operationName -> Operation
+	operations map[string]map[string]*schemaloader.Operation
```

Comment:
For read heavy ops, like this one, use sync.Map instead. I've done benchmarking for a similar use case and sync.Map is much better optimized for this use case compared to a map/map with mutex.

Another thought to consider:

If we create the map once and then "only" read from it, even concurrently, we don't need a mutex at all.
Depending on the architecture, it's possible to populate the map at creation time of the rpc handler and then never change it. Instead of modifying operations, you can swap the entire rpc handler and create a new registry. With this approach, no concurrency handling on read is required, so the code is simpler and much more efficient because we can remove locks.